components/brave_wallet_ui/components/extension/transaction-detail-panel/index.tsx

```diff
@@ -0,0 +1,230 @@
+import * as React from 'react'
+import * as EthereumBlockies from 'ethereum-blockies'
+import { useTransactionParser } from '../../../common/hooks'
+
+// Utils
+import { reduceAddress } from '../../../utils/reduce-address'
+import { getTransactionStatusString } from '../../../utils/tx-utils'
+import { toProperCase } from '../../../utils/string-utils'
+import { mojoTimeDeltaToJSDate } from '../../../utils/datetime-utils'
+import { formatFiatAmountWithCommasAndDecimals, formatWithCommasAndDecimals } from '../../../utils/format-prices'
+
+import { getLocale } from '../../../../common/locale'
+import {
+  BraveWallet,
+  WalletAccountType,
+  DefaultCurrencies
+} from '../../../constants/types'
+import Header from '../../buy-send-swap/select-header'
+
+// Styled Components
+import {
+  StyledWrapper,
+  OrbContainer,
+  FromCircle,
+  ToCircle,
+  DetailRow,
+  DetailTitle,
+  DetailButton,
+  StatusRow,
+  BalanceColumn,
+  TransactionValue,
+  PanelDescription,
+  SpacerText,
+  FromToRow,
+  AccountNameText,
+  ArrowIcon
+} from './style'
+
+import {
+  StatusBubble,
+  DetailTextDarkBold,
+  DetailTextDark
+} from '../shared-panel-styles'
+
+export interface Props {
+  transaction: BraveWallet.TransactionInfo
+  selectedNetwork: BraveWallet.EthereumChain
+  accounts: WalletAccountType[]
+  visibleTokens: BraveWallet.ERCToken[]
+  transactionSpotPrices: BraveWallet.AssetPrice[]
+  defaultCurrencies: DefaultCurrencies
+  onBack: () => void
+  onRetryTransaction: (transaction: BraveWallet.TransactionInfo) => void
+  onSpeedupTransaction: (transaction: BraveWallet.TransactionInfo) => void
+  onCancelTransaction: (transaction: BraveWallet.TransactionInfo) => void
+}
+
+const TransactionDetailPanel = (props: Props) => {
+  const {
+    transaction,
+    selectedNetwork,
+    accounts,
+    visibleTokens,
+    transactionSpotPrices,
+    defaultCurrencies,
+    onBack,
+    onRetryTransaction,
+    onSpeedupTransaction,
+    onCancelTransaction
+  } = props
+
+  const parseTransaction = useTransactionParser(selectedNetwork, accounts, transactionSpotPrices, visibleTokens)
+  const transactionDetails = React.useMemo(
+    () => parseTransaction(transaction),
+    [transaction]
+  )
+
+  const fromOrb = React.useMemo(() => {
+    return EthereumBlockies.create({ seed: transactionDetails.sender.toLowerCase(), size: 8, scale: 16 }).toDataURL()
+  }, [transactionDetails.sender])
+
+  const toOrb = React.useMemo(() => {
+    return EthereumBlockies.create({ seed: transactionDetails.recipient.toLowerCase(), size: 8, scale: 16 }).toDataURL()
+  }, [transactionDetails.recipient])
+
+  const onClickViewOnBlockExplorer = () => {
+    const explorerURL = selectedNetwork.blockExplorerUrls[0]
```

Comment:
Not for this PR, but I think we should have this method associated with the `EthereumChain` type (not sure how it's done in typescript). We have this logic duplicated in many places.

Reply:
Issue created here! https://github.com/brave/brave-browser/issues/20192